### PR TITLE
Feature/#17 make style for html generation adjustable

### PR DESCRIPTION
**Pull request by waltersve**

### CODE REVIEW

**Comment by jpradocueva** [comment]
@waltersve, @BirgitBoss this Pull Request contains at least one file with an MIT license. The Working Group agreed on its charger that the output of this group will be released under the Mozilla Public License v2.0 (MPL 2.0). 

Despite being MPL 2.0 a weak copyleft license and MIT a more permissive license, I am not in a position to provide advice on the compatibility of these licenses.
My suggestion is that the Administrators, and/or Editors of this repository should be consulted before importing third-party code which is not under MLP 2.0. 
There are a few considerations when it comes to software licenses compatibility that should be reviewed by a license expert.

Even assuming that in this particular case there is no license incompatibility, the Organization team should be made aware, at the time of the ratification, of a mix of licenses in the repository.

**Comment by waltersve** [comment]
Hi @jpradocueva, MPL 2.0 is combatible with MIT see the offizial page here: https://www.mozilla.org/en-US/MPL/license-policy/ -> Licenses Compatible with the MPL. I hope that helps.

File: core/sds-aspect-model-document-generators/README.md [hunk 1]
```diff
@@ -31,23 +31,40 @@ For more information on SPARQL construct queries, Graphviz and Mustache see:
 ## Document Generator
 
 The document generator creates an HTML page for an Aspect Model. The page includes the diagram for the Aspect Model.
-
-Embedding font and images for making the html page to run standalone without dependencies to the internet 
-the following online service was used: https://www.font-converter.net/en/css-embedded-font-base64
+The HTML page embedding all required styles, fonts and images for making the html page to run standalone without
```
Comment (atextor) [inline core/sds-aspect-model-document-generators/README.md:34]: Use upper case HTML consistently (in the beginning of the line, it's "HTML", later it's "html")

File: core/sds-aspect-model-document-generators/README.md [hunk 2]
```diff
@@ -31,23 +31,40 @@ For more information on SPARQL construct queries, Graphviz and Mustache see:
...
 ### How it works
 
 For the generation the aspect java meta model is used.
 
 1. The Aspect Model definition file (.ttl) is loaded
 2. The aspect meta model is instantiated with the contents of the loaded aspect meta model definition
-3. The instantiated aspect meta model is used to generate Asciidoc using the `AspectModelAsciidoc.xtend` XTend template
-4. The resulting Asciidoc is converted into an HTML file using AsciidoctorJ
-5. The diagram for the Aspect Model is generated in SVG format and embedded into the HTML as a Base64 encoded String
+3. The instantiated aspect meta model is used to generate the HTMl page using Velocity templates
```
Comment (atextor) [inline core/sds-aspect-model-document-generators/README.md:43]: HTML

File: core/sds-aspect-model-document-generators/src/main/java/io/openmanufacturing/sds/aspectmodel/generator/AspectModelHelper.java [hunk 3]
```diff
@@ -0,0 +1,180 @@
+/*
```
Comment (atextor) [inline core/sds-aspect-model-document-generators/src/main/java/io/openmanufacturing/sds/aspectmodel/generator/AspectModelHelper.java:1]: All Java files do not seem to follow the code style used in the project. Please adjust.

File: core/sds-aspect-model-document-generators/src/main/java/io/openmanufacturing/sds/aspectmodel/generator/AspectModelHelper.java [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+    public static Set<Constraint> getConstraints(final Property property) {
+        final Set<Constraint> constraints = new HashSet<>();
+        if (property.getCharacteristic() instanceof Trait) {
+            constraints.addAll(((Trait) property.getCharacteristic()).getConstraints());
+        }
+        return constraints;
+    }
+
+    @SuppressWarnings("squid:S2250")
```
Comment (atextor) [inline core/sds-aspect-model-document-generators/src/main/java/io/openmanufacturing/sds/aspectmodel/generator/AspectModelHelper.java:109]: Please add a comment to the annotation that explains what is suppressed and why

File: core/sds-aspect-model-document-generators/src/main/java/io/openmanufacturing/sds/aspectmodel/generator/AspectModelHelper.java [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+                .orElseThrow(() -> new DocumentGenerationException("Could not find entity " + singleEntity
+                        + " in list of entities: " + entities));
+    }
+
+    public static String getNameFromURN(final String urn) {
+        String[] parts = urn.split("#");
+        return parts.length == 2 ? parts[1] : urn;
+    }
+
+    @SuppressWarnings("squid:S3655")
```
Comment (atextor) [inline core/sds-aspect-model-document-generators/src/main/java/io/openmanufacturing/sds/aspectmodel/generator/AspectModelHelper.java:149]: Please add a comment to the annotation that explains what is suppressed and why

File: core/sds-aspect-model-document-generators/src/main/java/io/openmanufacturing/sds/aspectmodel/generator/docu/AspectModelDocumentationGenerator.java [hunk 6]
```diff
@@ -13,320 +13,241 @@
...
+
+        @Override
+        public OutputStream apply(final String artifactName) {
+            final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+            artifacts.put(artifactName, outputStream);
+            return outputStream;
+        }
+    }
+
+    @SuppressWarnings("squid:S1166")
```
Comment (atextor) [inline core/sds-aspect-model-document-generators/src/main/java/io/openmanufacturing/sds/aspectmodel/generator/docu/AspectModelDocumentationGenerator.java:179]: Please add a comment to the annotation that explains what is suppressed and why

**Review by atextor — CHANGES_REQUESTED** [review]

**Comment by jpradocueva** [comment]
@waltersve I don't disagree, but unfortunately, I am not in a position to provide advice with regards to licensing. My main point is that it should be a mechanism built into the submission process that checks for any license other than MPL v2.0. 

A good reference that I normally use to evaluate compatibility is this one https://dwheeler.com/essays/floss-license-slide.html
I am happy to assist the Administrators and Editors with this task.

File: core/sds-aspect-model-document-generators/README.md (hunk 1 above)
Comment (waltersve) [reply core/sds-aspect-model-document-generators/README.md]: done

File: core/sds-aspect-model-document-generators/README.md (hunk 2 above)
Comment (waltersve) [reply core/sds-aspect-model-document-generators/README.md]: done

File: core/sds-aspect-model-document-generators/src/main/java/io/openmanufacturing/sds/aspectmodel/generator/AspectModelHelper.java (hunk 4 above)
Comment (waltersve) [reply core/sds-aspect-model-document-generators/src/main/java/io/openmanufacturing/sds/aspectmodel/generator/AspectModelHelper.java]: done

File: core/sds-aspect-model-document-generators/src/main/java/io/openmanufacturing/sds/aspectmodel/generator/AspectModelHelper.java (hunk 5 above)
Comment (waltersve) [reply core/sds-aspect-model-document-generators/src/main/java/io/openmanufacturing/sds/aspectmodel/generator/AspectModelHelper.java]: done

File: core/sds-aspect-model-document-generators/src/main/java/io/openmanufacturing/sds/aspectmodel/generator/docu/AspectModelDocumentationGenerator.java (hunk 6 above)
Comment (waltersve) [reply core/sds-aspect-model-document-generators/src/main/java/io/openmanufacturing/sds/aspectmodel/generator/docu/AspectModelDocumentationGenerator.java]: done

File: core/sds-aspect-model-document-generators/src/main/java/io/openmanufacturing/sds/aspectmodel/generator/AspectModelHelper.java (hunk 3 above)
Comment (waltersve) [reply core/sds-aspect-model-document-generators/src/main/java/io/openmanufacturing/sds/aspectmodel/generator/AspectModelHelper.java:1]: What do you mean. I changed only the name and added an additonal method?

Comment (waltersve) [reply core/sds-aspect-model-document-generators/src/main/java/io/openmanufacturing/sds/aspectmodel/generator/AspectModelHelper.java:1]: done

**Pull request merged**